test/drivers/test_revd2.cc

```diff
@@ -14,69 +14,94 @@ class TestREVD2 : public ::testing::Test
 
     virtual void TearDown() {};
 
-    /// General test for RSVD:
-    /// Computes the decomposition factors, then checks A-U\Sigma\transpose{V}.
-    template <typename T, typename RNG>
-    static void test_REVD2_general(int64_t m, int64_t k, int64_t k_start, std::tuple<int, T, bool> mat_type, RandBLAS::base::RNGState<RNG> state, int rank_expectation, T err_expectation) {
+    template <typename T>
+    struct REVD2TestData {
+        std::vector<T> A_buf;
+        std::vector<T> A;
+        std::vector<T> A_cpy;
+        std::vector<T> V;
+        std::vector<T> eigvals;
+        std::vector<T> E;
+        std::vector<T> Buf;
+
+        REVD2TestData(int64_t m, int64_t k) :
```

Comment:
It will help you later if you keep track of ``m`` and ``k`` in this struct.

Reply:
Not sure if i understand what you mean by this.

Reply:
If ``testdata`` is an ``REVD2TestData`` struct, then it should have fields ``testdata.m, testdata.k``. That way you don't need to pass around those integer parameters separately from ``REVD2TestData`` object.

Reply:
Yes, but every time we use them now we'll have to write testdata.m, which is long and messes with readability. Plus techically those are not data, just variables.

Reply:
But if you think that this is the right way, i'll do it.

Reply:
Restating from another thread:

> You can unpack dimension parameters on entry to a function, like
```
auto m = testdata.m;
auto n = testdata.n;
auto k = testdata.k;
```

Reply:
So then all this extra code would serve the sole purpose of passing one object instead of object and variables?

Reply:
Resolved.